thrift.js

```diff
@@ -277,58 +276,32 @@ Thrift.prototype.compileService = function compileService(def) {
     var service = new ThriftService({strict: self.strict});
     service.compile(def, self);
     self.define(service.name, def.id, service);
-    self.services[service.name] = service;
 };
 
-Thrift.prototype.compileConst = function compileConst(def, spec) {
+Thrift.prototype.compileConst = function compileConst(def, model) {
     var self = this;
     var thriftConst = new ThriftConst(def);
     self.define(def.id.name, def.id, thriftConst);
-    self.consts[def.id.name] = thriftConst;
 };
 
 Thrift.prototype.compileEnum = function compileEnum(def) {
     var self = this;
-    var spec = new ThriftEnum();
-    spec.compile(def, self);
-    self.define(spec.name, def.id, spec);
-    self.enums[spec.name] = spec;
-    self.types[spec.name] = spec;
+    var model = new ThriftEnum();
+    model.compile(def, self);
+    self.define(model.name, def.id, model);
 };
 
 Thrift.prototype.link = function link() {
     var self = this;
-    var index;
 
     if (self.linked) {
         return self;
     }
     self.linked = true;
 
-    var moduleNames = Object.keys(self.modules);
-    for (index = 0; index < moduleNames.length; index++) {
-        var thriftModule = self.modules[moduleNames[index]];
-        thriftModule.link(self);
-        self.modules[moduleNames[index]] = thriftModule;
-    }
-
-    var typeNames = Object.keys(self.types);
-    for (index = 0; index < typeNames.length; index++) {
-        var type = self.types[typeNames[index]];
-        self[type.name] = type.link(self).surface;
-    }
-
-    var serviceNames = Object.keys(self.services);
-    for (index = 0; index < serviceNames.length; index++) {
-        var service = self.services[serviceNames[index]];
-        self[service.name] = service.link(self).surface;
-    }
-
-    var constNames = Object.keys(self.consts);
-    for (index = 0; index < constNames.length; index++) {
-        var thriftConst = self.consts[constNames[index]];
-        self.consts[constNames[index]] = thriftConst.link(self);
-        self[thriftConst.name] = thriftConst.link(self).surface;
+    var names = Object.keys(self.models);
+    for (var index = 0; index < names.length; index++) {
+        self.models[names[index]].link(self);
```

Comment:
Moving the surface logic into Thrift model link methods makes it possible to just enumerate the models instead of special cases for each kind of model.